### ISSUE TITLE
Exampe outbox is not workering

### ISSUE THREAD

**Issue by bjartek** [vgarvardt/gue#114]
Hello, 

I tried the example merged in #98 but i get an error when i run the client task. After starting the client and putting messages in the db i run the `task client` comamnd

I get the following error:
```
 task worker
task: [deps-up] docker compose up --detach --wait
[+] Running 2/2
 ⠿ Container outbox-worker-kafka-redpanda-1  Healthy                                                                                                                                      0.5s
 ⠿ Container outbox-worker-kafka-postgres-1  Healthy                                                                                                                                      0.5s
task: [_start_worker] go run ./... worker
Error: could not create test topic: dial tcp [::1]:9092: connect: connection refused
```

Looking in docker logs i see
`outbox-worker-kafka-postgres-1 | 2022-10-17 12:17:08.136 UTC [76] FATAL:  role "test" does not exist`

**Comment by vgarvardt** [comment]
Just tested with the fresh setup - works fine. Do you have `DB_DSN` env var set? Because role `test` is not coming from outbox, it is `DB_DSN: 'postgres://outbox:outbox@localhost:{{.PG_PORT}}/outbox?sslmode=disable'`

**Comment by bjartek** [comment]
Just tested this on another computer as well, and I get the exact same results. Both computers are m1 macs. 

I run:

- task worker, add messages, verify that they are in the table
- in another termianl run task client and get the error you see above. 

I do not have the DB_DSN var set outside of the taskfile.

**Comment by bjartek** [comment]
The imageId of images used on my current machine (laptop) is:
Postgres: 84497fa250d0
Redpanda 15e88ba3e219

**Comment by vgarvardt** [comment]
Added a little bit of debug logs at https://github.com/vgarvardt/gue/commit/81b78d98ee2bc318c678cdc87fd4e26322e24259 - can you pls try to run it and share logs.

**Comment by bjartek** [comment]
```task client
task: [deps-up] docker compose up --detach --wait
[+] Running 2/2
 ⠿ Container outbox-worker-kafka-redpanda-1  Running                                       0.0s
 ⠿ Container outbox-worker-kafka-postgres-1  Running                                       0.0s
task: [_start_client] go run ./... client
2022/10/18 22:58:33 Connecting to the DB "postgres://outbox:outbox@localhost:62497/outbox?sslmode=disable"
? How many message should I publish to kafka using outbox? 10
2022/10/18 22:58:37 Tried to enqueue a job level=debug client-id=outbox-worker-client-cfdeca error=<nil> queue=outbox-kafka id=21
2022/10/18 22:58:37 Enqueued message for publishing to the gue: topic "test-topic", key "2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584-0", value "message #0 @ 2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584"
2022/10/18 22:58:37 Tried to enqueue a job level=debug client-id=outbox-worker-client-cfdeca error=<nil> queue=outbox-kafka id=22
2022/10/18 22:58:37 Enqueued message for publishing to the gue: topic "test-topic", key "2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584-1", value "message #1 @ 2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584"
2022/10/18 22:58:37 Tried to enqueue a job level=debug client-id=outbox-worker-client-cfdeca error=<nil> queue=outbox-kafka id=23
2022/10/18 22:58:37 Enqueued message for publishing to the gue: topic "test-topic", key "2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584-0", value "message #2 @ 2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584"
2022/10/18 22:58:37 Tried to enqueue a job level=debug client-id=outbox-worker-client-cfdeca error=<nil> queue=outbox-kafka id=24
2022/10/18 22:58:37 Enqueued message for publishing to the gue: topic "test-topic", key "2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584-1", value "message #3 @ 2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584"
2022/10/18 22:58:37 Tried to enqueue a job level=debug client-id=outbox-worker-client-cfdeca error=<nil> queue=outbox-kafka id=25
2022/10/18 22:58:37 Enqueued message for publishing to the gue: topic "test-topic", key "2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584-0", value "message #4 @ 2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584"
2022/10/18 22:58:37 Tried to enqueue a job level=debug client-id=outbox-worker-client-cfdeca error=<nil> queue=outbox-kafka id=26
2022/10/18 22:58:37 Enqueued message for publishing to the gue: topic "test-topic", key "2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584-1", value "message #5 @ 2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584"
2022/10/18 22:58:37 Tried to enqueue a job level=debug client-id=outbox-worker-client-cfdeca error=<nil> queue=outbox-kafka id=27
2022/10/18 22:58:37 Enqueued message for publishing to the gue: topic "test-topic", key "2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584-0", value "message #6 @ 2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584"
2022/10/18 22:58:37 Tried to enqueue a job level=debug client-id=outbox-worker-client-cfdeca error=<nil> queue=outbox-kafka id=28
2022/10/18 22:58:37 Enqueued message for publishing to the gue: topic "test-topic", key "2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584-1", value "message #7 @ 2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584"
2022/10/18 22:58:37 Tried to enqueue a job level=debug client-id=outbox-worker-client-cfdeca error=<nil> queue=outbox-kafka id=29
2022/10/18 22:58:37 Enqueued message for publishing to the gue: topic "test-topic", key "2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584-0", value "message #8 @ 2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584"
2022/10/18 22:58:37 Tried to enqueue a job level=debug client-id=outbox-worker-client-cfdeca error=<nil> queue=outbox-kafka id=30
2022/10/18 22:58:37 Enqueued message for publishing to the gue: topic "test-topic", key "2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584-1", value "message #9 @ 2022-10-18 22:58:37.313117 +0200 CEST m=+3.773202584"
? How many message should I publish to kafka using outbox? (10)
```

```
task worker
task: [deps-up] docker compose up --detach --wait
[+] Running 2/2
 ⠿ Container outbox-worker-kafka-postgres-1  Running                                       0.0s
 ⠿ Container outbox-worker-kafka-redpanda-1  Running                                       0.0s
task: [_start_worker] go run ./... worker
2022/10/18 22:59:16 Connecting to the DB "postgres://outbox:outbox@localhost:62497/outbox?sslmode=disable"
2022/10/18 22:59:16 Initialising test kafka topic at "localhost:62495"
Error: could not create test topic: dial tcp [::1]:9092: connect: connection refused
Usage:
  outbox worker [flags]

Flags:
  -h, --help   help for worker

2022/10/18 22:59:16 could not create test topic: dial tcp [::1]:9092: connect: connection refused
exit status 1
task: Failed to run task "worker": task: Failed to run task "_start_worker": exit status 1
```

**Comment by vgarvardt** [comment]
> Error: could not create test topic: dial tcp [::1]:9092: connect: connection refused

This error is very strange - according to logs redpanda is running on port `62495`, but kafka admin is still trying to connect to 9092. Will check this part deeper.

**Comment by bjartek** [comment]
Anyways the generic pattern of this example is very useful to me. :)

**Comment by vgarvardt** [comment]
Ok, found out where the issue is - Kafka client connects to the broker exposed on the random port and requests metadata. Among this metadata there is an advertised address that has fixed port, so client tries to establish connection to it as well and fails. The only solution I found so far is to expose fixed port for Kafka. Fixed at https://github.com/vgarvardt/gue/commit/a4b35fa2e01342573c8454818ac742d443b23ded

PS Kafka has really good naming - every time I work with it a little bit deeper than simple publishing/consuming from pre-configured cluster it feels like reading Kafka books - same feelings of senselessness, hopelessness, absurd...
